### PR TITLE
Fix padding and font sizes on side menu texts

### PR DESCRIPTION
**Pull request by aurangzaibumer**
configured side menu end text color and font size along with app version font size on the SIDE PANEL

image (26)
Fixes : https://github.com/onaio/fhir-resources/issues/3360

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by aurangzaibumer** [comment]
@HenryRae @AnnieMungai please check the above screenshot for UI changes and provide feedback. 
thanks

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/app/ApplicationConfiguration.kt [hunk 1]
```diff
@@ -28,6 +28,7 @@ data class ApplicationConfiguration(
   override var configType: String = ConfigType.Application.name,
   val appTitle: String = "",
   val remoteSyncPageSize: Int = 100,
+  val appVersionFontSize: Float? = null,
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/app/ApplicationConfiguration.kt:31]: @ellykits @HenryRae Can we have defaults for this?

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/navigation/NavigationMenuConfig.kt [hunk 2]
```diff
@@ -33,6 +33,8 @@ data class NavigationMenuConfig(
   val showCount: Boolean = false,
   val animate: Boolean = true,
   val actions: List<ActionConfig>? = null,
+  val countFontSize: Float? = null,
+  val countColor: String? = null,
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/navigation/NavigationMenuConfig.kt:37]: @ellykits @HenryRae can we have defaults for this?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 3]
```diff
@@ -375,11 +385,11 @@ private fun SideMenuItem(
 }
 
 @Composable
-private fun SideMenuItemText(title: String, textColor: Color) {
+private fun SideMenuItemText(title: String, textColor: Color, fontSize: Float? = null) {
   Text(
     text = title,
     color = textColor,
-    fontSize = 18.sp,
+    fontSize = fontSize?.sp ?: 18.sp,
```
Comment (aurangzaibumer) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:392]: we are using default value if fontSize is null @dubdabasoduba

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 4]
```diff
@@ -144,6 +148,8 @@ fun AppDrawer(
             title = navigationMenu.display,
             endText = appUiState.registerCountMap[navigationMenu.id]?.toString() ?: "",
             showEndText = navigationMenu.showCount,
+            endTextColor = navigationMenu.countColor?.parseColor() ?: Color.White,
```
Comment (aurangzaibumer) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:151]: we are using default color if countColor is null @dubdabasoduba

**Comment by HenryRae** [comment]
Updates reviewed and are okay. 
- Only app version text size remaining.
Cc: @aurangzaibumer

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 5]
```diff
@@ -151,10 +151,13 @@ fun AppDrawer(
...
       modifier = modifier.padding(innerPadding).background(SideMenuDarkColor).fillMaxSize(),
     ) {
-      LazyColumn(modifier = modifier.padding(horizontal = 16.dp)) {
+      LazyColumn(modifier = modifier) {
         item {
-          Column {
-            if (navigationConfiguration.clientRegisters.isNotEmpty()) {
+          Column(modifier = modifier.padding(horizontal = 16.dp)) {
+            if (
+              navigationConfiguration.clientRegisters.isNotEmpty() &&
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:158]: This is not empty check is not necessary

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 6]
```diff
@@ -172,7 +175,7 @@ fun AppDrawer(
             title = navigationMenu.display,
             endText = appUiState.registerCountMap[navigationMenu.id]?.toString() ?: "",
             showEndText = navigationMenu.showCount,
-            endTextColor = MenuItemColor,
+            endTextColor = AppTitleColor,
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:178]: Can you update the color code for the MenuItemColor instead

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 7]
```diff
@@ -183,7 +186,7 @@ fun AppDrawer(
 
         item {
           if (navigationConfiguration.bottomSheetRegisters?.registers?.isNotEmpty() == true) {
-            Column {
+            Column(modifier = modifier.padding(horizontal = 16.dp)) {
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:189]: Isn't this adding extra horizontal padding?

**Review by ellykits — CHANGES_REQUESTED** [review]

**Pull request merged**